org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/embedder/MavenExecutionContext.java

```diff
@@ -114,7 +126,16 @@ public MavenExecutionContext(IComponentLookup lookup, IMavenProjectFacade projec
       }
       this.basedir = basedir == null ? null : (basedir.isDirectory() ? basedir : basedir.getParentFile());
     } else {
-      this.basedir = null;
+      ClassWorld world = null;
```

Comment:
This looks suspicious what is this good for?

Reply:
In same cases, for example `ProjectRegistryManager.readProjectsWithDependencies()` the context is created without a project facade but the lookUp has a basedir and is not the global `M2EClassWorld`.
Because the context's basedir is later used to search the `multiModuleProjectDirectory` it is important to get that basedir. Otherwise the content of the `.mvn` folder is not considered. Unfortunately the result of that call is used in the workspace build.

Of course that could be done when creating the execution request and the basedir of the context is null.

![grafik](https://user-images.githubusercontent.com/44067969/199299979-ce60ebaf-326a-40d3-8506-f269fac68673.png)

Reply:
I think we should pass the basedir as a constructor parameter as well instead of read it from the project facade indirectly.

Reply:
> I think we should pass the basedir as a constructor parameter as well instead of read it from the project facade indirectly.

I have reworked the the constructors, but nevertheless I was not able to completely get rid of that indirect retrieval, although it is in more limited scope now.

Reply:
Then we maybe need to refactor tings here, I thing the execution should always be well defined in regard to lookup + basedir. I'll take a look and maybe try to prepare an extra PR as a prerequisite to this.

Reply:
Thanks for #1027 this part is now resolved.